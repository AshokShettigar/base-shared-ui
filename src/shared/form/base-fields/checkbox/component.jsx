import { PureComponent } from 'react';
import { string, bool, func, node } from 'prop-types';
import classNames from 'classnames';

import { FieldWithError } from '../index';
import { withRender } from '../../../shared';
import styles from './component.scss';
import { Icon } from '../../../icon';

class Checkbox extends PureComponent {
    static propTypes = {
        id: string,
        name: string,
        value: string,
        children: node,
        error: string,

        autoFocus: bool,
        readOnly: bool,
        checked: bool,
        multiple: bool,
        disabled: bool,
        indeterminate: bool,

        className: string,
        labelClassName: string,

        innerRef: func,
        onChange: func
    };

    static defaultProps = {
        id: undefined,
        name: undefined,
        value: undefined,
        children: undefined,
        error: undefined,

        autoFocus: false,
        readOnly: false,
        checked: false,
        multiple: false,
        disabled: false,
        indeterminate: false,

        className: undefined,
        labelClassName: undefined,

        innerRef: undefined,
        onChange: undefined
    };

    onChange = () => {
        if (this.props.onChange) {
            this.props.onChange({
                name: this.props.name,
                value: this.props.value,
                checked: this.checkboxRef.checked
            });
        }
    };

    checkboxRef = null;

    setRef = el => {
        this.checkboxRef = el;

        if (this.props.innerRef) {
            this.props.innerRef(el);
        }
    };

    render() {
        const className = classNames(styles.root, this.props.className, {
            [styles.multiple]: this.props.multiple,
            [styles.disabled]: this.props.disabled,
            [styles.readOnly]: this.props.readOnly,
            [styles.indeterminate]: this.props.indeterminate,
            [styles.checked]: this.props.checked
        });

        return (
            <span className={className}>
                <FieldWithError error={this.props.error}>
                    <label htmlFor={this.props.id} className={styles.label}>
                        <input
                            hidden
                            type="checkbox"
                            autoFocus={this.props.autoFocus} // eslint-disable-line
                            name={this.props.name}
                            value={this.props.value}
                            checked={this.props.checked}
                            disabled={this.props.disabled}
                            ref={this.setRef}
                            onChange={this.onChange}
                        />
                        <span className={styles.checkbox}>
                            {this.props.indeterminate && <i className={styles.indeterminateGlyph} />}
                            {this.props.checked && !this.props.indeterminate &&
                                <span className={styles.checkmarkGlyph}>
                                    <Icon glyph="checkmark" size="mini" className={styles.icon} />
                                </span>
                            }
                        </span>
                        {this.props.children &&
                            <span
                                tabIndex={-1}
                                role="radio"
                                aria-checked={this.props.checked}
                                className={classNames(styles.labelText, this.props.labelClassName)}
                            >
                                {this.props.children}
                            </span>
                    }
                    </label>
                </FieldWithError>
            </span>
        );
    }
}

export default withRender()(Checkbox);
