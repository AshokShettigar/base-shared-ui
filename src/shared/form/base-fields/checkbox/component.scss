@import '../../../scss/variables';

.root {
  display: inline-flex;
  font-family: $ff-open-sans;
  font-size: $fs-tiny;
  color: $c-raisin-black;
}

.label {
  display: flex;
  cursor: pointer;
  margin: 0;
}

.labelText {
  line-height: 12px;
  margin-left: 5px;
  color: $c-raisin-black;
  user-select: none;
  display: flex;
}

.checkbox {
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  min-width: 12px;
  height: 12px;
  border: 1px solid $c-gainsboro;
  border-radius: 2px;
  background-color: $c-white;
  z-index: 1;
}

.checkmarkGlyph,
.indeterminateGlyph {
  width: 10px;
  height: 10px;
  position: relative;

  &::after {
    position: absolute;
    top: 50%;
    left: 50%;
  }
}

.checkmarkGlyph {
  &::after {
    margin-top: -5px;
    margin-left: -5px;
  }
}

.indeterminateGlyph {
  &::after {
    content: ' ';
    width: 8px;
    height: 1px;
    background: currentColor;
    margin-left: -4px;
  }
}

.multiple {
  position: relative;

  &::after {
    content: ' ';
    position: absolute;
    width: 12px;
    height: 12px;
    bottom: -2px;
    left: -2px;
    border: 1px solid $c-gainsboro;
    border-top-color: transparent;
    border-right-color: transparent;
    display: block;
    border-radius: 2px;
  }

  &.checked::after {
    border-left: 1px solid $c-french-blue;
    border-bottom: 1px solid $c-french-blue;
  }
}

.checked, .indeterminate {
  .checkbox {
    background-color: $c-french-blue;
    border: 1px solid transparent;
    color: $c-white;
  }
}

.disabled, .readOnly {
  .label {
    cursor: default;
  }

  .checkbox {
    border: 1px solid $c-gainsboro;
    background-color: $c-white-smoke;
    color: $c-gainsboro;
  }
}

.icon {
  display: flex;
}
