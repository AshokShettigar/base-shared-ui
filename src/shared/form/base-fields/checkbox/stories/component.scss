.root {
  line-height: 1;
}

.item {
  margin-bottom: 5px;
  vertical-align: top;
}
