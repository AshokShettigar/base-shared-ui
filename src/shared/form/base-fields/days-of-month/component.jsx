import React, { PureComponent } from 'react';
import { string, bool, node, func, number, arrayOf } from 'prop-types';
import classNames from 'classnames';
import { v4 as uuid } from 'uuid';

import { FieldWithLabel, FieldWithError } from '../index';
import { DayCell } from '../shared';
import { withRender } from '../../../shared';
import styles from './component.scss';

class DaysOfMonth extends PureComponent {
    static propTypes = {
        id: string,
        error: node,
        label: string,
        name: string,
        value: arrayOf(number),
        description: string,

        required: bool,
        isDisabled: bool,

        className: string,
        labelClassName: string,
        cellClassName: string,

        onChange: func.isRequired
    };

    static defaultProps = {
        id: undefined,
        error: null,
        label: undefined,
        name: undefined,
        value: [],
        description: undefined,

        required: false,
        isDisabled: false,

        className: undefined,
        labelClassName: undefined,
        cellClassName: undefined,

        onChange: undefined
    };

    handleChange = day => {
        const daysOfMonth = new Set(this.props.value);

        if (daysOfMonth.has(day)) {
            daysOfMonth.delete(day);
        } else {
            daysOfMonth.add(day);
        }

        this.props.onChange(this.props.name, [...daysOfMonth]);
    };

    render() {
        return (
            <div className={classNames(styles.root, this.props.className)}>
                <FieldWithError error={this.props.error}>
                    <FieldWithLabel
                        required={this.props.required}
                        id={this.props.id}
                        label={this.props.label}
                        className={this.props.labelClassName}
                    >
                        <div className={styles.dateCard}>
                            {[...Array(31).keys()].map(day => (
                                <DayCell
                                    key={uuid()}
                                    className={classNames(
                                        styles.dayCell,
                                        {
                                            [styles.selected]: this.props.value.includes(day + 1),
                                            [styles.error]: this.props.error
                                        },
                                        this.props.cellClassName
                                    )}
                                    value={day + 1}
                                    label={`${day + 1}`}
                                    isDisabled={this.props.isDisabled}
                                    onChange={this.handleChange}
                                />
                            ))}
                        </div>
                    </FieldWithLabel>
                </FieldWithError>
                {this.props.description && (
                    <div className={styles.description}>
                        {this.props.description}
                    </div>
                )}
            </div>
        );
    }
}

export default withRender()(DaysOfMonth);
