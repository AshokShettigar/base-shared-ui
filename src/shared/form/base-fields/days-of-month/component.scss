@import '../../../scss/variables';

.root {
  display: flex;
  flex-direction: row;

  .description {
    display: inline-block;
    width: 350px;
    padding: 20px 15px;
    color: $c-raisin-black;
  }

  .dateCard {
    width: 210px;
  }

  .dayCell {
    width: 30px;
    height: 30px;
    box-shadow: none;
    border-top: none;
    border-left: none;

    &:first-child {
      border-top-left-radius: 3px;
    }

    &.selected {
      background-color: $c-north-texas-green;
      border-color: $c-white;
      color: $c-white;
    }

    &:nth-child(-n + 7) {
      border-top: 1px solid $c-gainsboro;

      &.selected {
        border-top: 1px solid $c-north-texas-green;
      }

      &.error {
        border-top: 1px solid $c-upsdell-red;
      }
    }

    &:nth-child(7n) {
      border-right: 1px solid $c-gainsboro;

      &.selected {
        border-right: 1px solid $c-north-texas-green;
      }

      &.error {
        border-right: 1px solid $c-upsdell-red;
      }
    }

    &:nth-child(n + 25) {
      border-bottom: 1px solid $c-gainsboro;

      &.selected {
        border-bottom: 1px solid $c-north-texas-green;
      }

      &.error {
        border-bottom: 1px solid $c-upsdell-red;
      }
    }

    &:nth-child(7n + 1) {
      border-left: 1px solid $c-gainsboro;

      &.selected {
        border-left: 1px solid $c-north-texas-green;
      }

      &.error {
        border-left: 1px solid $c-upsdell-red;
      }
    }

    &:nth-child(7) {
      border-top-right-radius: 3px;
    }

    &:nth-child(29) {
      border-bottom-left-radius: 3px;
    }

    &:last-child,
    &:nth-child(28) {
      border-bottom-right-radius: 3px;

      &.error {
        border-right: 1px solid $c-upsdell-red;
      }
    }

    &:focus {
      outline: none;
      box-shadow: 0 0 2px 1px $c-fresh-air;
    }
  }
}
