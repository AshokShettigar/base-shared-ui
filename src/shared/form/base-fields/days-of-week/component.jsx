import React, { PureComponent } from 'react';
import { string, bool, node, number, func, oneOf, arrayOf } from 'prop-types';
import classNames from 'classnames';
import { v4 as uuid } from 'uuid';

import { FieldWithLabel, FieldWithError } from '../index';
import { DayCell, DAYS_OF_WEEK } from '../shared';
import styles from './component.scss';
import { withRender } from '../../../shared/with-render';

class DaysOfWeek extends PureComponent {
    static propTypes = {
        id: string,
        name: string,
        label: string,
        format: oneOf(['short', 'long']),
        value: arrayOf(number),
        error: node,

        isRequired: bool,
        isDisabled: bool,

        className: string,
        labelClassName: string,
        cellClassName: string,

        onChange: func.isRequired
    };

    static defaultProps = {
        id: undefined,
        label: undefined,
        format: 'short',
        value: [],
        error: null,

        className: undefined,
        labelClassName: undefined,
        cellClassName: undefined,

        isRequired: false,
        isDisabled: false,

        onChange: undefined
    };

    handleChange = day => {
        const daysOfWeek = new Set(this.props.value);

        if (daysOfWeek.has(day)) {
            daysOfWeek.delete(day);
        } else {
            daysOfWeek.add(day);
        }

        this.props.onChange(this.props.name, [...daysOfWeek]);
    };

    render() {
        return (
            <div className={classNames(styles.root, this.props.className)}>
                <FieldWithError error={this.props.error}>
                    <FieldWithLabel
                        id={this.props.id}
                        label={this.props.label}
                        required={this.props.isRequired}
                        className={this.props.labelClassName}
                    >
                        <div className={styles.daysList}>
                            {DAYS_OF_WEEK.map(day => (
                                <DayCell
                                    key={uuid()}
                                    isDisabled={this.props.isDisabled}
                                    className={classNames(
                                    styles.dayCell,
                                        {
                                            [styles.selected]: this.props.value.includes(day.number),
                                            [styles.error]: this.props.error
                                        },
                                        this.props.cellClassName
                                )}
                                    value={day.number}
                                    label={this.props.format === 'short' ? day.shortName : day.name}
                                    onChange={this.handleChange}
                                />
                        ))}
                        </div>
                    </FieldWithLabel>
                </FieldWithError>
            </div>
        );
    }
}

export default withRender()(DaysOfWeek);
