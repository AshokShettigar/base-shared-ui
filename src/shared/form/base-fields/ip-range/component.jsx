import React, { Component } from 'react';
import { string, bool, node, func, objectOf } from 'prop-types';
import { isEmpty } from 'lodash';
import classNames from 'classnames';

import { KEY_CODES, isKeyCodeNumber } from '../../../helpers';
import { FieldWithError, FieldWithLabel, IPField } from '../index';
import { withRender } from '../../../shared';
import styles from './component.scss';

export class IPv4RangeField extends Component {
    static propTypes = {
        label: string,
        name: objectOf(string),
        value: objectOf(string).isRequired,
        error: node,
        startIPRangeError: node,
        endIPRangeError: node,
        separator: node,

        autoFocus: bool,
        disabled: bool,
        readOnly: bool,
        required: bool,
        hidden: bool,
        autoWidth: bool,
        hideErrorMessages: bool,

        className: string,
        labelClassName: string,
        fieldWrapperClassName: string,
        ipFieldClassName: string,
        fieldClassName: string,
        contentClassName: string,
        errorClassName: string,

        onChange: func.isRequired,
        onBlur: func
    };

    static defaultProps = {
        name: {
            start: undefined,
            end: undefined
        },
        value: {
            start: undefined,
            end: undefined
        },
        error: undefined,
        separator: undefined,

        autoFocus: false,
        disabled: false,
        readOnly: false,
        required: false,
        hidden: false,
        autoWidth: false,
        hideErrorMessages: false,

        className: undefined,
        labelClassName: undefined,
        fieldWrapperClassName: undefined,
        fieldClassName: undefined,
        contentClassName: undefined
    };

    componentDidMount () {
        if (this.props.autoWidth) {
            this.resizeInput(this.startIPFieldRef);
            this.resizeInput(this.endIPFieldRef);
        }
    }

    setStartIPFieldInnerRef = ref => {
        this.startIPFieldRef = ref;
    };

    setEndIPFieldInnerRef = ref => {
        this.endIPFieldRef = ref;
    };

    getFieldsValues = () => ({
        start: this.startIPFieldRef.value,
        end: this.endIPFieldRef.value
    });

    startIPFieldRef = null;

    endIPFieldRef = null;

    tabKeys = [KEY_CODES.ARROW_RIGHT, KEY_CODES.SPACE, KEY_CODES.DASH, KEY_CODES.TAB];

    backKeys = [KEY_CODES.ARROW_LEFT, KEY_CODES.BACKSPACE];

    isTabKeyPressed = keyCode => this.tabKeys.includes(keyCode);

    isBackKeyPressed = keyCode => this.backKeys.includes(keyCode);

    hasThreeDots = value => value.split('.').length - 1 === 3;

    isAllowedFocusEndField = (e, isValid) => {
        const { keyCode, target: { value, selectionStart: caretPosition } } = e;
        const isEndCaretPosition = caretPosition === value.length;

        return !isValid
            && e.target === this.startIPFieldRef
            && isEndCaretPosition
            && this.hasThreeDots(value)
            && (this.isTabKeyPressed(keyCode) || isKeyCodeNumber(keyCode));
    };

    isAllowedFocusStartField = e => {
        const { keyCode, target: { selectionStart, selectionEnd } } = e;
        const isStartCaretPosition = selectionStart === selectionEnd && selectionStart === 0;

        return e.target === this.endIPFieldRef
            && isStartCaretPosition
            && this.isBackKeyPressed(keyCode);
    };

    isAllowedSetEndFieldValue = ({ keyCode }) => isKeyCodeNumber(keyCode) && isEmpty(this.endIPFieldRef.value);

    resizeInput = ({ value, style }) => {
        const symbolsAmount = value.length;

        style.width = symbolsAmount ? `${(symbolsAmount * 7) + 20}px` : '120px';
    };

    handleKeyDown = (e, isValid) => {
        if (this.isAllowedFocusEndField(e, isValid)) {
            this.endIPFieldRef.focus();

            if (this.isAllowedSetEndFieldValue(e)) {
                this.endIPFieldRef.value = e.key;
            }

            e.preventDefault();
        }

        if (this.isAllowedFocusStartField(e)) {
            this.startIPFieldRef.focus();
            e.preventDefault();
        }
    };

    handleChange = ({ target }) => {
        if (this.props.autoWidth) {
            this.resizeInput(target);
        }

        this.props.onChange(this.getFieldsValues());
    };

    handleBlur = () => {
        if (this.props.onBlur) {
            this.props.onBlur(this.getFieldsValues());
        }
    };

    render() {
        return (
            <FieldWithError
                className={classNames(styles.root, this.props.errorClassName)}
                hideErrorMessages={this.props.hideErrorMessages}
                error={this.props.error}
            >
                <FieldWithLabel
                    label={this.props.label}
                    required={this.props.required}
                    className={this.props.labelClassName}
                >
                    <div className={classNames(styles.fields, this.props.className)}>
                        <IPField
                            name={this.props.name.start}
                            value={this.props.value.start}
                            error={this.props.startIPRangeError}

                            autoFocus={this.props.autoFocus}
                            disabled={this.props.disabled}
                            readOnly={this.props.readOnly}
                            required={this.props.required}

                            className={this.props.fieldWrapperClassName}
                            fieldClassName={classNames(this.props.fieldClassName, {
                                [styles.autoWidth]: !isEmpty(this.props.value.start) && this.props.autoWidth
                            })}

                            innerRef={this.setStartIPFieldInnerRef}
                            onBlur={this.handleBlur}
                            onKeyDown={this.handleKeyDown}
                            onKeyUp={this.handleKeyUp}
                            onChange={this.handleChange}
                        />
                        {
                            this.props.separator
                                ? this.props.separator
                                : <span className={styles.separator}>-</span>
                        }
                        <IPField
                            name={this.props.name.end}
                            value={this.props.value.end}
                            error={this.props.endIPRangeError}

                            autoFocus={false}
                            disabled={this.props.disabled}
                            readOnly={this.props.readOnly}
                            required={this.props.required}

                            className={this.props.fieldWrapperClassName}
                            fieldClassName={classNames(this.props.fieldClassName, {
                                [styles.autoWidth]: !isEmpty(this.props.value.end) && this.props.autoWidth
                            })}

                            innerRef={this.setEndIPFieldInnerRef}
                            onBlur={this.handleBlur}
                            onKeyDown={this.handleKeyDown}
                            onKeyUp={this.handleKeyUp}
                            onChange={this.handleChange}
                        />
                    </div>
                </FieldWithLabel>
            </FieldWithError>
        );
    }
}

export default withRender()(IPv4RangeField);
