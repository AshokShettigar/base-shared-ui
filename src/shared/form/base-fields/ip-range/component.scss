@import '../../../scss/variables';

.root {
}

.separator {
  margin: 5px 5px;
}

.fields {
  display: flex;
  flex-direction: row;
}

.autoWidth {
  min-width: 40px;
  text-align: center;
}
