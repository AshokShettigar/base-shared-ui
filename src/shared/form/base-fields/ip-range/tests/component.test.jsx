import { shallow } from 'enzyme';

import { KEY_CODES } from '../../../../helpers';
import { IPv4RangeField } from '../component';

const defaultProps = {
    label: 'label',
    value: {
        start: null,
        end: null
    },
    onChange: jest.fn(),
    onBlur: jest.fn()
};

describe('IPv4RangeField Component', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<IPv4RangeField {...defaultProps} />);
    });

    it('should render component with default props', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should return true', () => {
        const actual = wrapper.instance().isTabKeyPressed(KEY_CODES.ARROW_RIGHT);

        expect(actual).toBeTruthy();
    });

    it('should return false', () => {
        const actual = wrapper.instance().isBackKeyPressed(1);

        expect(actual).toBeFalsy();
    });

    describe('hasThreeDots', () => {
        it('should return true', () => {
            const actual = wrapper.instance().hasThreeDots('123.123.123.123');

            expect(actual).toBeTruthy();
        });

        it('should return false', () => {
            const actual = wrapper.instance().hasThreeDots('123.123.');

            expect(actual).toBeFalsy();
        });
    });

    it('should allow focusing to the second field', () => {
        const target = {
            value: '123.123.123.123',
            selectionStart: 15
        };
        const e = {
            keyCode: KEY_CODES.ARROW_RIGHT,
            target
        };
        wrapper.instance().startIPFieldRef = target;
        const actual = wrapper.instance().isAllowedFocusEndField(e, false);

        expect(actual).toBeTruthy();
    });

    it('should allow focusing to the first field', () => {
        const target = {
            value: '',
            selectionStart: 0,
            selectionEnd: 0
        };
        const e = {
            keyCode: KEY_CODES.ARROW_LEFT,
            target
        };
        wrapper.instance().endIPFieldRef = target;
        const actual = wrapper.instance().isAllowedFocusStartField(e);

        expect(actual).toBeTruthy();
    });

    it('should allow set value to the second field', () => {
        const e = {
            keyCode: 48
        };
        wrapper.instance().endIPFieldRef = {
            value: ''
        };
        const actual = wrapper.instance().isAllowedSetEndFieldValue(e);

        expect(actual).toBeTruthy();
    });

    describe('handleKeyDown()', () => {
        it('should trigger the isAllowedFocusEndField method', () => {
            const e = {
                keyCode: KEY_CODES.ARROW_RIGHT,
                key: 1,
                target: {
                    value: 'value',
                    selectionStart: 0
                },
                preventDefault: jest.fn()
            };
            wrapper.instance().endIPFieldRef = {
                focus: jest.fn()
            };
            wrapper.instance().isAllowedFocusEndField = jest.fn(() => true);
            wrapper.instance().isAllowedSetEndFieldValue = jest.fn(() => true);
            wrapper.instance().handleKeyDown(e, true);

            expect(wrapper.instance().isAllowedFocusEndField).toHaveBeenCalled();
            expect(wrapper.instance().endIPFieldRef.focus).toHaveBeenCalled();
            expect(wrapper.instance().isAllowedSetEndFieldValue).toHaveBeenCalled();
            expect(e.preventDefault).toHaveBeenCalled();
        });

        it('should trigger the isAllowedFocusStartField method', () => {
            const e = {
                keyCode: KEY_CODES.ARROW_LEFT,
                target: {
                    value: 'value',
                    selectionStart: 0
                },
                preventDefault: jest.fn()
            };
            wrapper.instance().startIPFieldRef = {
                focus: jest.fn()
            };
            wrapper.instance().isAllowedFocusStartField = jest.fn(() => true);
            wrapper.instance().handleKeyDown(e, true);

            expect(wrapper.instance().isAllowedFocusStartField).toHaveBeenCalled();
            expect(wrapper.instance().startIPFieldRef.focus).toHaveBeenCalled();
            expect(e.preventDefault).toHaveBeenCalled();
        });
    });

    it('should trigger the onChange method', () => {
        const target = {
            value: 'value',
            selectionStart: 0
        };
        const e = {
            target
        };
        wrapper.instance().getFieldsValues = jest.fn();
        wrapper.instance().handleChange(e);

        expect(defaultProps.onChange).toHaveBeenCalled();
    });

    it('should trigger the onBlur method', () => {
        wrapper.instance().getFieldsValues = jest.fn();
        wrapper.instance().handleBlur();

        expect(defaultProps.onBlur).toHaveBeenCalled();
    });

    it('should set the startIPFieldRef ref', () => {
        const ref = 'ref';
        wrapper.instance().setStartIPFieldInnerRef(ref);

        expect(wrapper.instance().startIPFieldRef).toEqual(ref);
    });

    it('should set the endIPFieldRef ref', () => {
        const ref = 'ref';
        wrapper.instance().setEndIPFieldInnerRef(ref);

        expect(wrapper.instance().endIPFieldRef).toEqual(ref);
    });

    it('should get values of fields', () => {
        const value = 'value';
        wrapper.instance().startIPFieldRef = {
            value
        };
        wrapper.instance().endIPFieldRef = {
            value
        };
        const actual = wrapper.instance().getFieldsValues();

        expect(actual).toEqual({ start: value, end: value });
    });
});
