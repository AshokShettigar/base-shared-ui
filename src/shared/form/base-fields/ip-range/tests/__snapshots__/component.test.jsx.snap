// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IPv4RangeField Component should render component with default props 1`] = `
<WithRender
  className="root"
  hidden={false}
  hideErrorMessages={false}
  rendered={true}
>
  <WithRender
    hidden={false}
    label="label"
    rendered={true}
    required={false}
  >
    <div
      className="fields"
      data-component="IPv4RangeField"
    >
      <IPv4Field
        autoFocus={false}
        disabled={false}
        fieldClassName=""
        hidden={false}
        hideErrorMessages={false}
        innerRef={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        readOnly={false}
        required={false}
        value={null}
      />
      <span
        className="separator"
      >
        -
      </span>
      <IPv4Field
        autoFocus={false}
        disabled={false}
        fieldClassName=""
        hidden={false}
        hideErrorMessages={false}
        innerRef={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        readOnly={false}
        required={false}
        value={null}
      />
    </div>
  </WithRender>
</WithRender>
`;
